### ISSUE TITLE
variables user_account_name and user_full_name do not work in Qfield

### ISSUE THREAD

**Issue by bakkerss** [opengisch/QField#729]
I like to store the userID by default when data is stored so I am able to check who entered data. In the desktop version of my project I tried using the variables 'user_account_name' and 'user_full_name' which worked fine. The operation system's userID is returned (windows).
However, in Qfield the same field is not defauted with the userID. Instead the error "not available"  is shown".

1. Project is copied to Tablet and opened in Qfield
2. I open the layer to add new sightings.
3. The user field displays "not available" 

### Expected behavior:

I expect the user account name from the tablet's OS (Android) is collected.

### Observed behavior:

Error message "not available" is reported

### Screenshots and GIFs

(https://github.com/opengisch/QField/files/3898032/Qfield.issue.userID.pdf)

**QField version:** 1.2.0 - Matterhorn (10200997)

**Comment by bakkerss** [comment]
It seems part of the issue had been caused by an issue in the preceding fields. Once this issue had been fixed the variable was displaying a value. 

However a new problem now appears. The user field shows the value u0_a179. This seems to be the technical value of the account user. Can this be changed to the description of the account user (so the normal name of the person)?

**Comment by signedav** [comment]
Thanks for reporting. I need to investigate here.

**Comment by signedav** [comment]
What do you have on QField in Settings - Variables?
![Screenshot_20191128-172903_QField](https://user-images.githubusercontent.com/28384354/69821753-d4591b00-1204-11ea-9112-d0cde9371355.jpg)

**Comment by bakkerss** [comment]
Attached is the screenshot of the variables.

![Screenshot Qfield](https://user-images.githubusercontent.com/45461027/69874534-92d77700-12bb-11ea-82ef-134b226e87b9.jpeg)

It is similar to yours. I would have expected to see the account user name, in this case my personal name.

**Comment by signedav** [comment]
`user_full_name` is never available on Android because it does not exists. The `user_account_name` is apparently the account id of the app (QField).

There might be a possibility to get more useful data instead as these variables, but I don't really know what should be taken, since the device has not a user like the desktop PC has.

Or am I talking wrong stuff @m-kuhn?

Of course you could add the variable you need in QField on each device. But possibly you don't want the field workers to change something in the settings and you don't want to configure everyones tablet neither. So hm... maybe there could be a feature request to enter a username on startup. Just thinking...

**Comment by bakkerss** [comment]
Hi Signedav,

This might be a usefull link:  https://developer.android.com/training/id-auth/identify

Personally, I wouldn't mind adding a name in the Q-field settings which can be used. When you share a device, this will allow you to use multiple usernames. Furthermore, if the description is equal to a username in QGIS tables you are for example able to link personal data like telephone numbers etc. This is useful if you want to contact the field workers. 

But I think this is up to you guys to decide on the best solution.

Kind regards,

Sebastiaan

**Comment by mbernasocchi** [comment]
As a workaround you could define a global variable

**Comment by m-kuhn** [comment]
Yes, global variables is the currently implemented and recommended way.

But Integration of AccountManager sounds even better. I tagged this as an enhancement, if anyone would like to sponsor this, we'll be more than happy to work on it.

**Comment by nirvn** [comment]
Let's close this now that we have @cloud_username and @cloud_useremail exposed as a global variable (since QField 4.1.8). For non-cloud usages, people can definitively still go ahead and manually add a global variable (or these days, do it via plugins too 😸 )
